gnosis/eth/oracles/oracles.py

```diff
@@ -231,6 +256,22 @@ def __init__(
             ethereum_client.w3, self.router_address
         )
 
+    @classmethod
+    def is_available(
+        cls,
+        ethereum_client: EthereumClient,
+    ) -> bool:
+        """
+        :param ethereum_client:
+        :return: `True` if Oracle is available for the EthereumClient provided, `False` otherwise
+        """
+        return ethereum_client.is_contract(
+            cls.ROUTER_ADDRESSES.get(
+                ethereum_client.get_network(),
+                cls.ROUTER_ADDRESSES[EthereumNetwork.MAINNET],
+            )
```

Comment:
Why do we default to `Mainnet` in for `UniswapV2Oracle` but not for `UniswapOracle`?

Reply:
Because `Uniswap` used deterministic deployment for `V2` but not for `V1`